app/models/post.rb

```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  def update_user_posts_count
+    author.update(posts_count: author.posts.count)
+  end
+
```

Comment:
- **[OPTIONAL]** It would be advisable to add an `after_save` active record callback for this method. This callback ensure that this method is called whenever a post is saved, so that the author's post counter is updated without having to call this method directly.

  ```
  after_save :update_user_posts_count
  ...

  private

  def update_user_posts_count
    author.update(posts_count: author.posts.count)
  end
  ```